rust/private/utils.bzl

```diff
@@ -379,3 +388,80 @@ def transform_deps(deps, make_rust_providers_target_independent):
         build_info = dep[BuildInfo] if BuildInfo in dep else None,
         cc_info = dep[CcInfo] if CcInfo in dep else None,
     ) for dep in deps] if make_rust_providers_target_independent else deps
+
+def is_third_party_package(package, third_party_dir):
+    """Determines if the given package is considered third-party.
+
+    Args:
+        package (string): The package in question.
+        third_party_dir (string): The directory in which third-party packages are kept.
+
+    Returns:
+        True if the package is considered third-party, False otherwise.
+    """
+    return ("//" + package + "/").startswith(third_party_dir + "/")
```

Comment:
Please add `TODO(hlopko): This code assumes a monorepo; make it work with external repositories as well.

Reply:
Done. Would supporting external repositories be as simple as checking if the package is in a different repo, and not renaming in that case? If so, I can add that case in this PR.

Reply:
I think supporting external repos means mangling the repo name into the crate name, and checking the repo in the label of the third party package, and writing a bunch of tests :)